### PR TITLE
[PDI-16323] Ability to publish data source over SSL to the BA server

### PR DESCRIPTION
**Pull request by Advent51**
- secure check was improved

### CODE REVIEW

**Comment by Advent51** [comment]
@pamval @stepanovdg please review

**Comment by stepanovdg** [comment]
Have no permissions to merge in this project.

**Comment by mchen-len-son** [comment]
@pamval @mkambol please take a look.
thanks.

File: pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinConnection.java [hunk 1]
```diff
@@ -556,7 +556,9 @@ public ThinConnection build() throws SQLException {
     }
 
     private URI baseUri() throws SQLException {
-      boolean isSecure = "true".equalsIgnoreCase( arguments.get( ARG_ISSECURE ) );
+      boolean isSecure = "true".equalsIgnoreCase( arguments.get( ARG_ISSECURE ) )
+              || uri.getPort() == 8443
```
Comment (pamval) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinConnection.java:560]: Is this really the only way to know if a connection is secure? This looks very error prone.

Comment (mchen-len-son) [reply pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinConnection.java:560]: probably a more reliable option would be: `"https".equalsIgnoreCase(` uri.getScheme() )`.
WDYT, @pamval ?

Comment (pamval) [reply pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinConnection.java:560]: yup, agree

**Comment by mchen-len-son** [comment]
This one should be closed in favor of #106.
@pamval can you please close it?

**Pull request closed without merging**